qiskit_experiments/library/characterization/resonator_spectroscopy.py

```diff
@@ -0,0 +1,185 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Spectroscopy experiment class for resonators."""
+
+from typing import Iterable, Optional, Tuple
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.exceptions import QiskitError
+from qiskit.providers import Backend
+import qiskit.pulse as pulse
+
+from qiskit_experiments.library.characterization import QubitSpectroscopy
+from qiskit_experiments.data_processing.processor_library import get_processor, ProjectorType
+from .analysis.resonator_spectroscopy_analysis import ResonatorSpectroscopyAnalysis
+
+
+class ResonatorSpectroscopy(QubitSpectroscopy):
+    """Perform spectroscopy on the readout resonator.
+
+    # section: overview
+        This experiment does spectroscopy on the readout resonator. It applies the following
+        circuit
+
+        .. parsed-literal::
+
+                 ┌─┐
+              q: ┤M├
+                 └╥┘
+            c: 1/═╩═
+                  0
+
+        where a spectroscopy pulse is attached to the measurement instruction. When doing
+        readout resonator spectroscopy, each measured IQ point has a frequency dependent
+        phase. Close to the resonance, the IQ points start rotating around in the IQ plan.
+        This effect must be accounted for in the data processing to produce a meaningful
+        signal.
+
+    # section: analysis_ref
+        :py:class:`ResonatorSpectroscopyAnalysis`
+
+    # section: see_also
+        qiskit_experiments.library.characterization.qubit_spectroscopy.QubitSpectroscopy
+    """
+
+    def __init__(
+        self,
+        qubit: int,
+        frequencies: Iterable[float],
+        backend: Optional[Backend] = None,
+        absolute: bool = True,
+        **experiment_options,
+    ):
+        """
+        A spectroscopy experiment run by setting the frequency of the readout drive.
+        The parameters of the GaussianSquare spectroscopy pulse can be specified at run-time.
+        The spectroscopy pulse has the following parameters:
+        - amp: The amplitude of the pulse must be between 0 and 1, the default is 0.1.
```

Comment:
From a usability perspective, I think it would be best to default to `amp=1` since this measurement does not require the qubit to be operational and more power will give more SNR. Also, timing values in seconds would be easier to work with here than samples.

Reply:
This takes care of the amp: dfe3149. Seconds would be easier to work with. However, this requires two things:
* the `dt` which we have as `backend.configuration().dt` 
* the sample multiple which for us is 16. Currently, I believe that we do not have this at all in our configuration. This makes it hard to avoid hard-coding checks that the schedules are multiples of 16.

Reply:
Correction: we have the needed info in `timing_constraints`.

Reply:
This is also done. We can now give the experiment in units of `s` or `dt`.